src/core/lib/resolver/resolver_factory.h

```diff
@@ -54,9 +54,8 @@ class ResolverFactory {
  public:
   virtual ~ResolverFactory() {}
 
-  /// Returns the URI scheme that this factory implements.
-  /// Caller does NOT take ownership of result.
-  virtual absl::string_view scheme() const = 0;
+  /// Returns true if the factory implements \a scheme, false otherwise.
+  virtual bool ImplementsScheme(absl::string_view scheme) const = 0;
```

Comment:
It looks like the changes to name resolver registration are losing the ability to tell if we've registered multiple resolvers for the same scheme.  I would prefer not to lose that check.

What was the goal of this change?  Was it just to allow a single name resolver to support multiple schemes?  If so, I think a better approach would be to have the resolver factory return a set of scheme names and then register the factory under each one.

Reply:
Doing so ties name resolver registration to address resolver registration in pretty ugly ways - and would likely force bespoke logic into config resolver just for name resolution.

By doing this we get to completely separate resolver registration from address parser registration, which I believe to be a worthwhile tradeoff.